scripts/CEE/list-alerts/script.sh

```diff
@@ -1,114 +1,159 @@
 #!/bin/bash
 
-# default is retrived all firing alerts (critical+warning)
+# default is retrieved all firing alerts (critical+warning)
 IS_CRITICAL=1
 IS_WARNING=1
 PRINT_SILENCE=0
 NAMESPACE_QUERY="namespace=~\"^$|^default$|^openshift-.*|^kube-.*|^redhat-.*\",namespace!~\"^redhat-rhmi-.*\""
 ALERTSTATE_QUERY="alertstate=\"firing\","
+declare -A hostarray
 
-ARG_ARRAY=( $SCRIPT_PARAMETERS )
-
-for ARG in "${ARG_ARRAY[@]}" ; do
-    if [ "$ARG" == "--warning-only" ] ; then
-        IS_CRITICAL=0
-        IS_WARNING_ONLY=1
-    elif [ "$ARG" == "--critical-only" ] ; then
-        IS_WARNING=0
-        IS_CRITICAL_ONLY=1
-    elif [ "$ARG" == "--pending-only" ] ; then
-        ALERTSTATE_QUERY="alertstate=\"pending\","
-        IS_PENDING_ONLY=1
-    elif [ "$ARG" == "--firing-only" ] ; then
-        ALERTSTATE_QUERY="alertstate=\"firing\","
-        IS_FIRING_ONLY=1
-    elif [ "$ARG" == "--all-states" ] ; then
-        ALERTSTATE_QUERY=""
-        NAMESPACE_QUERY=""
-        IS_ALL_STATES=1
-    elif [ "$ARG" == "--list-silences" ] ; then
-        PRINT_SILENCE=1
-    else
-        echo "usage: script.sh [--warning-only|--critical-only] [--firing-only|--pending-only|--all-states]"
-        echo "  --warning-only: print warning alerts only"
-        echo "  --critical-only: print critical alerts only"
-        echo "  --firing-only: print only firing alert state (default)"
-        echo "  --pending-only: print only pending alert state"
-        echo "  --all-states: print firing and pending alerts"
-        echo "  --list-silences: print all active silences"
-        exit 1
-    fi
+# Lists Prometheus hosts on the cluster if the --hosts argument is passed
+function list_prometheus_hosts(){
+	eval "oc get routes --all-namespaces -o custom-columns=NAMESPACE:.metadata.namespace,NAME:.metadata.name | grep prometheus"
+	exit 1
+}
+
+# Check if the namespaces provided exist
+function check_namespace_exists(){
+	PROMETHEUS_HOSTS_ARRAY=("$NAMESPACES")
+
+# If a namespace in the list doesn't exist, remove it
+for NAMESPACE in "${PROMETHEUS_HOSTS_ARRAY[@]}" ; do
+	if [[ $(oc get project "${NAMESPACE}" --no-headers) == "" ]]; then
+		unset PROMETHEUS_HOSTS_ARRAY[$NAMESPACE]
```

Comment:
While testing , I got this 
`/script.sh: line 24: unset: `PROMETHEUS_HOSTS_ARRAY[]': not a valid identifier `

even though, the functionality is being fulfilled : 
```
`./script.sh: line 24: unset: `PROMETHEUS_HOSTS_ARRAY[]': not a valid identifier
Retrieving only firing alerts
Retrieving both Critical and Warning alerts
Retrieving alerts from the prometheus-k8s Prometheus instance in the openshift-monitoring namespace:
Critical Alerts:
{}`
```
but this can be handled better.

Reply:
Thanks for pointing this out. Do you remember under what conditions you encountered this? I.e, did you export a list of namespaces to check? I've tried to reproduce but was unable.